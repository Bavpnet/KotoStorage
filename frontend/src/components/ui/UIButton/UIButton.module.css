.btn {
  background-color: black;
  color: white;
  border-radius: 10px;
  justify-content: center;
  align-items: center;
  display: flex;
}

.btn_text {
  font-size: 1rem;
  font-weight: 400;
  padding: 8px 30px;
}
