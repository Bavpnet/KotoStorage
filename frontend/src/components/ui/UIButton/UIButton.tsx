import styles from './UIButton.module.css'

type UIButtonProps = {
  text?: string
  onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void
  children?: React.ReactNode
}

export const UIButton: React.FC<UIButtonProps> = ({
  text,
  onClick,
  children
}) => (
  <button className={styles.btn} onClick={onClick}>
    {children ? children : <span className={styles.btn_text}>{text}</span>}
  </button>
)
