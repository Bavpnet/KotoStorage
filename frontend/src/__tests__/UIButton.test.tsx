import renderer from 'react-test-renderer'
import { UIButton } from '../components/ui/UIButton'

describe('UIButton', () => {
  it('renders correctly', () => {
    const text = 'test'
    const tree = renderer.create(<UIButton text={text} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
