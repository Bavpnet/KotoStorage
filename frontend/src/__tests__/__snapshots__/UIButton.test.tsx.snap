// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UIButton renders correctly 1`] = `
<button
  className="btn"
>
  test
</button>
`;
